### DISCUSSION TITLE
2.0.0-beta.2

### DISCUSSION THREAD

**Discussion by MichalLytek** [MichalLytek/type-graphql#1455] [General]
### Features
- **Breaking Change**: `AuthChecker` type is now "function or class" - update to `AuthCheckerFn` if the function form is needed in the code
- **Breaking Change**: update `graphql-js` peer dependency to `^16.6.0`
- **Breaking Change**: `buildSchemaSync` is now also checking the generated schema for errors
- **Breaking Change**: `validate` option of `buildSchema` is set to `false` by default - integration with `class-validator` has to be turned on explicitly
- **Breaking Change**: `validate` option of `buildSchema` doesn't accept anymore a custom validation function - use `validateFn` option instead
- support class-based auth checker, which allows for dependency injection
- allow defining directives for interface types and theirs fields, with inheritance for object types fields (#744)
- allow deprecating input fields and args (#794)
- support disabling inferring default values (#793)
- support readonly arrays for roles of `@Authorized` decorator (#935)
- add sync version of `buildTypeDefsAndResolvers` function (#803)
- lift restriction of listing all interfaces from inheritance chain in `implements` option of `@ObjectType` decorator (#1435)
### Fixes
- **Breaking Change**: properly emit types nullability when `defaultValue` is provided and remove `ConflictingDefaultWithNullableError` error (#751)
- allow defining extension on field resolver level for fields also defined as a property of the class (#776)
- fix throwing error when schema with dynamic default value was built again (#787)
- fix converting inputs with fields of nested array type (#801)
- disable broken exposing input types fields under a changed name via `@Field({ name: "..." })`
- support overwriting fields of extended types (#1109)
- properly execute args validation for nullable items array (#1328)
### Others
- **Breaking Change**: update `class-validator` peer dependency to `>=0.14.0`
- **Breaking Change**: change build config to ES2019 - drop support for Node.js < 14.5
- **Breaking Change**: remove support for loading resolvers by glob paths (`resolvers: string[]` build schema option)
- **Breaking Change**: remove `isAbstract` legacy decorator option

<hr /><em>This discussion was created from the release <a href='https://github.com/MichalLytek/type-graphql/releases/tag/v2.0.0-beta.2'>2.0.0-beta.2</a>.</em>

**Comment by MichalLytek** [comment]
Please report here all the issues encountered using this beta release 😉

**Reply by AChangXD** [reply]
Thanks Michal!

**Comment by hlspablo** [comment]
Awesome Michal

**Comment by chrisbruford** [comment]
Is there any documentation around the removal of `isAbstract`? the above notes it's "legacy" but I can still find numerous instances of it being the recommended approach?

https://prisma.typegraphql.com/docs/basics/usage/
https://typegraphql.com/docs/generic-types.html#basic-usage

**Reply by MichalLytek** [reply]
If you check the latest docs:
https://typegraphql.com/docs/next/generic-types.html#basic-usage
The `isAbstract` is removed from the code snippets.

`isAbstract` was related to the "build all from storage" approach before 1.0 release and have no impact when you not used string path-based `resolvers` option while building schema.

**Comment by fliitor** [comment]


**Reply by MichalLytek** [reply]
Missing args means it's an issue with your custom compiler - next.js, babel, swc, esbuild, tsx, etc.

**Reply by MichalLytek** [reply]
How the hell
```
@Arg("record", of => `${TItemClass.name}InputUpdate`)
```
 was working for you? You can't put string names of the types, but JS class references.

**Comment by bnussman** [comment]
2.0.0-beta.2 has been working solid in production for https://github.com/bnussman/beep! 🎉

**Comment by thongxuan** [comment]
we are still find isAbstract useful when you want to multiple decorators on a single field, for example

//-- in one file
@ObjectType({isAbstract: true})
@InputType()
class Item {
}

class Collection {
@Field({nullable: true})
items?: Item[];
}

//-- in other files
import {Item as ItemBase} from "./one";

@ObjectType()
export class Item extends ItemBase {
    @Directive('some_directive')
    some_other_field: SomeOtherClass;
}

**Reply by MichalLytek** [reply]
There should be no changes between having `isAbstract` and not. In `v1.x` releases `isAbstract` was ignored for `resolvers: [Foo, Bar]`.
